### PR TITLE
DPC-3973 login page

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3973

## 🛠 Changes

- Login Page Component added

## ℹ️ Context for reviewers

Based on https://designsystem.digital.gov/templates/authentication-pages/create-account/

## ✅ Acceptance Validation

Lookbook, automated and manual testing
Screenshot 2024-04-02 at 12 12 59 PM

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

**Review by ashley-weaver — APPROVED** [review]

**Review by kyeah — COMMENTED** [review]
Thanks! looks good.

Part of the A/C is updating the routing to remove MainController and route directly to the organizations page (which should pop open the login screen if needed) — is that implemented here?

File: dpc-portal/app/components/page/session/login_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<section class="grid-container usa-section">
```
Comment (kyeah) [inline dpc-portal/app/components/page/session/login_component.html.erb:1]: Not sure if we need grid-container here? The layout itself already has grid-container so this actually creates more margin than needed.

If you remove this then the login box aligns with the header icon:

CleanShot 2024-04-03 at 17 56 02@2x

File: dpc-portal/app/components/page/session/login_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <h1 class="margin-bottom-0">Sign in to your account</h1>
+        <p>You can access or create your account by signing in below.</p>
+        <%= button_to @login_path, class: 'usa-button width-full margin-bottom-3', data: { turbo: false } do %>
+          Sign in with <span class="login-button__logo">Login.gov</span>
+        <% end %>
+      </div>
+    </div>
+    <div class="grid-col-12 mobile-lg:grid-col-10 tablet:grid-col-8 desktop:grid-col-6 padding-x-205">
+      <div class="border-top border-base-lighter padding-top-4 desktop:border-0 desktop:padding-top-0">
+        <h2 class="display-none desktop:display-block">Welcome to the production portal for the  Data at the Point of Care (DPC) API</h2>
```
Comment (kyeah) [inline dpc-portal/app/components/page/session/login_component.html.erb:14]: Did you intend to hide this heading on mobile? 🤔 

Also small note that the heading includes top-margin by default. may want to explicitly set the margin-top to zero here so it aligns with the top of the login box and matches the designs a little more closely.

**Review by kyeah — COMMENTED** [review]
(adding two small design notes/questions)

File: dpc-portal/app/components/page/session/login_component.html.erb (hunk 2 above)
Comment (jdettmannnava) [reply dpc-portal/app/components/page/session/login_component.html.erb:14]: I took the classes directly from the example page; uswds is pretty complex so I like to do what they say.

Comment (jdettmannnava) [reply dpc-portal/app/components/page/session/login_component.html.erb]: I added the header that shows on mobile and set the desktop header to margin-top-0

**Review by kyeah — APPROVED** [review]
🌟

**Pull request merged**